### PR TITLE
Add sample for Vector Search and update README

### PR DESCRIPTION
**Pull request by ShivangiReja**
This PR introduces the following updates:

* Includes new samples demonstrating Vector Search functionality.
* Expands the test suite to cover additional scenarios related to Vector Search.
* Enhances the README by adding an example of authentication with Azure Active Directory (AAD).
* Provides additional information about Semantic Search and Vector Search in the README documentation.
* Fixes - https://github.com/Azure/azure-sdk-for-net/issues/37164

### CODE REVIEW

File: sdk/search/Azure.Search.Documents/samples/Sample07_VectorSearch.md [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+{
+    Console.WriteLine($"Answer Highlights: {result.Highlights}");
+    Console.WriteLine($"Answer Text: {result.Text}");
+}
+
+await foreach (SearchResult<Hotel> result in response.GetResultsAsync())
+{
+    count++;
+    Hotel doc = result.Document;
+    Console.WriteLine($"\nHotelId: {doc.HotelId} \n HotelName: {doc.HotelName} \n Category: {doc.Category} \n Description: {doc.Description}");
```
Comment (heaths) [inline sdk/search/Azure.Search.Documents/samples/Sample07_VectorSearch.md:272]: Long lines (also line 250 above) don't render well and make the customer scroll horizontally, which can be a pain with many mice. I'd use separate `Console.WriteLine` calls e.g.,
```
Console.WriteLine();
Console.WriteLine($"HotelId: {doc.HotelId}");
Console.WriteLine($" HotelName: {doc.HotelName}");
...
```
Also, was the single space before each property except the initial `HotelId` intentional?

File: sdk/search/Azure.Search.Documents/tests/Samples/Readme.cs [hunk 2]
```diff
@@ -49,6 +54,25 @@ public async Task Authenticate()
             #endregion Snippet:Azure_Search_Tests_Samples_Readme_Authenticate
         }
 
+        [Test]
+        [SyncOnly]
+        public async Task AuthenticateWithAAD()
+        {
+            await using SearchResources resources = await SearchResources.GetSharedHotelsIndexAsync(this, true);
+            Environment.SetEnvironmentVariable("SEARCH_ENDPOINT", resources.Endpoint.ToString());
```
Comment (heaths) [inline sdk/search/Azure.Search.Documents/tests/Samples/Readme.cs:62]: Tests run in parallel (from different assemblies) - maybe not important now, but you should avoid setting env vars that other code uses. Instead, consider using your actual search endpoint for the real test (probably from your `TestEnvironment`), and use `SNIPPET` symbol only for snippets, e.g.,

```c#
#if SNIPPET
Uri endpoint = new(Environment.GetEnvironmentVariable("SEARCH_ENDPOINT"));
#else
Uri endpoint = new(TestEnvironment.SearchEndpoint);
#endif
```

**Review by heaths — APPROVED** [review]
A few suggestions, but otherwise LGTM.

**Review by pallavit — APPROVED** [review]

File: sdk/search/Azure.Search.Documents/samples/Sample07_VectorSearch.md (hunk 1 above)
Comment (ShivangiReja) [reply sdk/search/Azure.Search.Documents/samples/Sample07_VectorSearch.md]: Updated the Console.WriteLine to maintain the consistency with the other sample codes.

**Pull request merged**